crepecake/src/main/java/com/google/cloud/tools/crepecake/registry/RegistryClient.java

```diff
@@ -74,7 +87,36 @@ public void pushManifest(V22ManifestTemplate manifestTemplate, String imageTag)
   public Blob pullBlob(DescriptorDigest blobDigest, Path destPath)
       throws RegistryException, IOException {
     BlobPuller blobPuller = new BlobPuller(registryEndpointProperties, blobDigest, destPath);
-    return callRegistryEndpoint(null, blobPuller);
+    return callRegistryEndpoint(blobPuller);
+  }
+
+  // TODO: Add mount with 'from' parameter
+  /**
+   * Pushes the BLOB, or skips if the BLOB already exists on the registry.
+   *
+   * @param blobDigest the digest of the BLOB, used for existence-check
+   * @param blob the BLOB to push
+   * @return {@code true} if the BLOB already exists on the registry and pushing was skipped; false
+   *     if the BLOB was pushed
+   */
+  public boolean pushBlob(DescriptorDigest blobDigest, Blob blob)
+      throws IOException, RegistryException {
+    BlobPusher blobPusher = new BlobPusher(registryEndpointProperties, blobDigest, blob);
+
+    // POST /v2/<name>/blobs/uploads/?mount={blob.digest}
+    String locationString = callRegistryEndpoint(blobPusher.initializer());
+    if (locationString == null) {
+      return true;
+    }
+    URL location = new URL(locationString);
```

Comment:
Can you make location or locationString more descriptive? is it a registryBlobPushTargetLocation? or something?

Reply:
Yea, it is the `Location` header to push the BLOB to. I added some comments to BlobPusher about the process. Would `patchLocation` (and `putLocation` below) be better?